kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3OutputStream.java

```diff
@@ -110,7 +116,9 @@ private void uploadPart(int size) throws IOException {
     }
 
     try {
-      multiPartUpload.uploadPart(new ByteArrayInputStream(buffer.array()), size);
+      InputStream inputStream = compressionType.wrapForInput(
+          buffer, RecordBatch.CURRENT_MAGIC_VALUE, BufferSupplier.NO_CACHING);
```

Comment:
The second and third parameters here look to be ignored for most CompressionType implementations.

I'm mostly focused on the use case of gzip, which I know ignores these two. The gzip format also is tolerant of concatenating several individually zipped streams to a single file. I am not certain that is the case for the other compression types. If that's not true, this approach of zipping each part of the multipart upload separately won't work. If it makes more sense to scope this to gzip and not support the other formats, I'm open to making that change.